node-hub/dora-phi4/dora_phi4/main.py

```diff
@@ -0,0 +1,148 @@
+import io
+from urllib.request import urlopen
+
+import pyarrow as pa
+import requests
+import soundfile as sf
+import torch
+from accelerate import infer_auto_device_map
+from dora import Node
+from PIL import Image
+from transformers import (
+    AutoModelForCausalLM,
+    AutoProcessor,
+    GenerationConfig,
+)
+
+# 🔍 Detect the best available device
+if torch.cuda.is_available():
+    device = "cuda"
+    torch_dtype = torch.float16  # Use float16 for efficiency
+elif torch.backends.mps.is_available():
+    device = "mps"
+    torch_dtype = torch.float16  # Reduce memory usage for MPS
+else:
+    device = "cpu"
+    torch_dtype = torch.float32  # CPU uses float32
+
+
+# Load the model and processor
+MODEL_PATH = "microsoft/Phi-4-multimodal-instruct"
+
+processor = AutoProcessor.from_pretrained(MODEL_PATH, trust_remote_code=True)
+# bnb_config = BitsAndBytesConfig(load_in_4bit=True, bnb_4bit_compute_dtype=torch.float16)
+
+model = AutoModelForCausalLM.from_pretrained(
+    MODEL_PATH,
+    # quantization_config=bnb_config,
+    torch_dtype=torch.float16
+    if device == "cuda"
+    else torch.bfloat16,  # Use bfloat16 for CPU
```

Comment:
```suggestion
    torch_dtype=torch_dtype
```

Seems to me that otherwise the type are inconssistent.

Reply:
Fixed!